src/equations/equations.jl

```diff
@@ -202,6 +202,84 @@ struct BoundaryConditionNeumann{B}
   boundary_normal_flux_function::B
 end
 
+"""
+    BoundaryConditionCoupled(other_semi_index, indices, uEltype)
+
+Boundary condition to glue two meshes together. Solution values at the boundary
+of another mesh will be used as boundary values. This requires the use
+of [`SemidiscretizationCoupled`](@ref). The other mesh is specified by `other_semi_index`,
+which is the index of the mesh in the tuple of semidiscretizations.
+
+Note that the elements and nodes of the two meshes at the coupled boundary must coincide.
+This is currently only implemented for [`StructuredMesh`](@ref).
+
+# Arguments
+- `other_semi_index`: the index in `SemidiscretizationCoupled` of the semidiscretization
+                      from which the values are copied
+- `indices::Tuple`: node/cell indices at the boundary of the mesh in the other
+                    semidiscretization. See examples below.
+- `uEltype::Type`: element type of solution
+
+# Examples
+```julia
+# Connect the left boundary of mesh 2 to our boundary such that our positive
+# boundary direction will match the positive y direction of the other boundary
+BoundaryConditionCoupled(2, (1, :i), Float64)
+
+# Connect the same two boundaries oppositely oriented
+BoundaryConditionCoupled(2, (1, :i_backwards), Float64)
+
+# Using this as y_neg boundary will connect `our_cells[i, 1, j]` to `other_cells[j, end-i, end]`
+BoundaryConditionCoupled(2, (:j, :i_backwards, :end), Float64)
+```
+
+!!! warning "Experimental code"
+    This is an experimental feature and can change any time.
+"""
+mutable struct BoundaryConditionCoupled{NDIMS, NDIMST2M1, uEltype<:Real, I}
+  # Buffer for boundary values: [variable, nodes_i, nodes_j, cell_i, cell_j]
+  u_boundary       ::Array{uEltype, NDIMST2M1} # NDIMS * 2 - 1
+  other_semi_index ::Int
+  other_orientation::Int
+  indices          ::I
```

Comment:
```suggestion
mutable struct BoundaryConditionCoupled{NDIMS, NDIMST2M1, uEltype<:Real, Indices}
  # Buffer for boundary values: [variable, nodes_i, nodes_j, cell_i, cell_j]
  u_boundary       ::Array{uEltype, NDIMST2M1} # NDIMS * 2 - 1
  other_semi_index ::Int
  other_orientation::Int
  indices          ::Indices
```
There is `I` from LinearAlgebra so it may be nice to avoid confusion